### PR TITLE
Remove do_repair option from citus_copy_shard_placement

### PR DESCRIPTION
**Pull request by marcocitus**
DESCRIPTION: Remove do_repair option from citus_copy_shard_placement

Remove shard repair functionality and adjust tests as part of post-11.0 cleanup.

### CODE REVIEW

File: src/backend/distributed/operations/shard_transfer.c [hunk 1]
```diff
@@ -212,99 +182,7 @@ citus_copy_shard_placement(PG_FUNCTION_ARGS)
...
 {
-	return citus_copy_shard_placement(fcinfo);
-}
-
-
-/*
- * ShardListSizeInBytes returns the size in bytes of a set of shard tables.
- */
-uint64
-ShardListSizeInBytes(List *shardList, char *workerNodeName, uint32
```
Comment (marcocitus) [inline src/backend/distributed/operations/shard_transfer.c:223]: moved to improve the structure of this file a bit (have UDFs at the top)

File: src/backend/distributed/sql/udfs/citus_copy_shard_placement/latest.sql [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
 	shard_id bigint,
 	source_node_name text,
 	source_node_port integer,
 	target_node_name text,
 	target_node_port integer,
-	do_repair bool DEFAULT true,
 	transfer_mode citus.shard_transfer_mode default 'auto')
 RETURNS void
-LANGUAGE C STRICT
+LANGUAGE C
```
Comment (JelteF) [inline src/backend/distributed/sql/udfs/citus_copy_shard_placement/latest.sql:10]: Removing STRICT seems unintended

File: src/backend/distributed/operations/shard_transfer.c [hunk 3]
```diff
@@ -170,37 +154,23 @@ citus_copy_shard_placement(PG_FUNCTION_ARGS)
...
-	if (doRepair && shardReplicationMode == TRANSFER_MODE_FORCE_LOGICAL)
-	{
-		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("logical replication cannot be used for repairs")));
-	}
 
 	ShardInterval *shardInterval = LoadShardInterval(shardId);
 	ErrorIfTableCannotBeReplicated(shardInterval->relationId);
 	ErrorIfTargetNodeIsNotSafeToCopyTo(targetNodeName, targetNodePort);
+	EnsureNoModificationsHaveBeenDone();
```
Comment (JelteF) [inline src/backend/distributed/operations/shard_transfer.c:167]: Why is this needed now?

File: src/backend/distributed/operations/shard_transfer.c [hunk 4]
```diff
@@ -212,99 +182,7 @@ citus_copy_shard_placement(PG_FUNCTION_ARGS)
...
-		ereport(ERROR, (errmsg("not enough empty space on node if the shard is moved, "
-							   "actual available space after move will be %ld bytes, "
-							   "desired available space after move is %ld bytes,"
-							   "estimated size increase on node after move is %ld bytes.",
-							   diskAvailableInBytesAfterShardMove,
-							   desiredNewDiskAvailableInBytes, colocationSizeInBytes),
-						errhint(
-							"consider lowering citus.desired_percent_disk_available_after_move.")));
-	}
+	ereport(ERROR, (errmsg("master_copy_shard_placement is deprecated")));
```
Comment (JelteF) [inline src/backend/distributed/operations/shard_transfer.c:185]: Let's still forward to citus_copy_shard_placement. Hard-deprecating all master_xxx UDFS seems like it would be better as a separate PR.

**Comment by JelteF** [comment]
The addition of src/test/regress/expected/replicated_partitioned_tables.out seems unintended (I cannot comment on an empty file).

File: src/backend/distributed/operations/shard_transfer.c (hunk 3 above)
Comment (marcocitus) [reply src/backend/distributed/operations/shard_transfer.c:167]: this check was missing from regular shard copy, which was a bug revealed by rewriting one of the tests from doing repair to regular copy

File: src/backend/distributed/sql/udfs/citus_copy_shard_placement/latest.sql (hunk 2 above)
Comment (marcocitus) [reply src/backend/distributed/sql/udfs/citus_copy_shard_placement/latest.sql:10]: thanks, I was first experimenting with nullifying do_repair, but in the end did not like the dangling argument very much

**Review by JelteF — APPROVED** [review]
